### PR TITLE
165151158-remove redundant properties

### PR DESCRIPTION
**Pull request by c-troha**
Redundant data field existed, was removed from the entire API

### CODE REVIEW

**Comment by escalonn** [comment]
get latest changes from broker, and give this PR a better name

File: src/Housing.Forecast.Service/Controllers/ForecastController.cs [hunk 1]
```diff
@@ -423,7 +423,7 @@ private Snapshot CreateSnapshot(int rooms, int users, string location, DateTime?
             else
                 snapshot.Created = (DateTime)date;
 
-           
+
```
Comment (stormrubio3000) [inline src/Housing.Forecast.Service/Controllers/ForecastController.cs]: Why was this added? If you could delete all the extra unneeded white space and extra lines that would be great

File: src/Housing.Forecast.Context/Repos/SnapshotRepo.cs [hunk 2]
```diff
@@ -74,9 +74,9 @@ public async Task<IList<Snapshot>> GetByLocationAsync(DateTime datetime, string
         /// <returns>
         /// This method should return a list of all snapshots that fall within the date range and tied to the location provided.
         /// </returns>
-        public async Task<IList<Snapshot>> GetBetweenDatesAtLocationAsync(DateTime Start, DateTime End, string location)
+        public async Task<IList<Snapshot>> GetBetweenDatesAtLocationAsync(DateTime? Start, DateTime? End, string location)
         {
-            return await _context.Snapshots.Where(s => s.Created.Date >= Start.Date && s.Created.Date <= End.Date && s.Location.Equals(location)).ToListAsync();
+            return await _context.Snapshots.Where(s => s.Created.Value >= Start.Value && s.Created.Value <= End.Value && s.Location.Equals(location)).ToListAsync();
```
Comment (mattvalencia) [inline src/Housing.Forecast.Context/Repos/SnapshotRepo.cs]: Might want to move these long lines that go off-screen into multiple lines so we can read them.

File: src/Housing.Forecast.Context/Repos/SnapshotRepo.cs [hunk 3]
```diff
@@ -60,9 +60,9 @@ public async Task<IList<Snapshot>> GetBetweenDatesAsync(DateTime Start, DateTime
         /// <returns>
         /// This method should a list of all snapshots for the specified location that were craeted on the provided date that are stored in the database.
         /// </returns>
-        public async Task<IList<Snapshot>> GetByLocationAsync(DateTime datetime, string location)
+        public async Task<IList<Snapshot>> GetByLocationAsync(DateTime? datetime, string location)
         {
-            return await _context.Snapshots.Where(s => s.Created.Date == datetime.Date && s.Location.Equals(location)).ToListAsync();
+            return await _context.Snapshots.Where(s => s.Created.Value == datetime.Value && s.Location.Equals(location)).ToListAsync();
```
Comment (mattvalencia) [inline src/Housing.Forecast.Context/Repos/SnapshotRepo.cs]: Might want to move these long lines that go off-screen into multiple lines so we can read them.

File: src/Housing.Forecast.Service/Controllers/ForecastController.cs [hunk 4]
```diff
@@ -353,7 +353,7 @@ private async Task<List<Snapshot>> CreateSnapshots(DateTime? date, string locati
                     rooms = await _room.GetByDateAsync(date.Value);
                     users = await _user.GetByDateAsync(date.Value);
 
-                    Snapshot snapshot = CreateSnapshot(date.Value, rooms.Select(x => x.Occupancy).Sum(), users.Count, string.IsNullOrEmpty(location) ? "All" : text.ToTitleCase(location));
+                    Snapshot snapshot = CreateSnapshot(rooms.Select(x => x.Occupancy).Sum(), users.Count, string.IsNullOrEmpty(location) ? "All" : text.ToTitleCase(location), date.Value);
```
Comment (mattvalencia) [inline src/Housing.Forecast.Service/Controllers/ForecastController.cs]: Might want to move these long lines that go off-screen into multiple lines so we can read them.

File: src/Housing.Forecast.Service/Controllers/ForecastController.cs [hunk 5]
```diff
@@ -372,7 +372,7 @@ private async Task<List<Snapshot>> CreateSnapshots(DateTime? date, string locati
                             users = await _user.GetByLocationAsync(d, location);
                         }
 
-                        Snapshot snapshot = CreateSnapshot(d, rooms.Select(x => x.Occupancy).Sum(), users.Count, string.IsNullOrEmpty(location) ? "All" : text.ToTitleCase(location));
+                        Snapshot snapshot = CreateSnapshot(rooms.Select(x => x.Occupancy).Sum(), users.Count, string.IsNullOrEmpty(location) ? "All" : text.ToTitleCase(location), d);
```
Comment (mattvalencia) [inline src/Housing.Forecast.Service/Controllers/ForecastController.cs]: Might want to move these long lines that go off-screen into multiple lines so we can read them.

File: src/Housing.Forecast.Testing/Context/SnapshotRepoTests.cs [hunk 6]
```diff
@@ -52,7 +52,8 @@ public async Task GetByDate_WithValidDate_ReturnNonEmptyCollection()
                 Snapshot snapshot = _context.Snapshots.FirstOrDefault();
 
                 // Act
-                snapshots = await _snapshotRepository.GetByDateAsync(snapshot.Created);
+                snapshots = await _snapshotRepository.GetByDateAsync((DateTime)snapshot.Created);
+
```
Comment (mattvalencia) [inline src/Housing.Forecast.Testing/Context/SnapshotRepoTests.cs]: Whitespace?

File: src/Housing.Forecast.Context/Repos/RoomRepo.cs [hunk 7]
```diff
@@ -31,19 +31,19 @@ public async Task<IList<Room>> GetAsync()
...
         }
 
-        public async Task<IList<Room>> GetByDateAsync(DateTime datetime)
+        public async Task<IList<Room>> GetByDateAsync(DateTime? datetime)
         {
             return await _context.Rooms.Where(u =>
-                u.Created.Date <= datetime.Date
-                    && (!u.Deleted.HasValue || u.Deleted.Value.Date > datetime.Date)).ToListAsync();
+                u.Created.Date <= datetime.Value
+                    && (!u.Deleted.HasValue || u.Deleted.Value.Date > datetime.Value)).ToListAsync();
```
Comment (escalonn) [inline src/Housing.Forecast.Context/Repos/RoomRepo.cs]: nullable types have auto-unboxing ability that usually makes this type of code unnecessary but hey
